src/values.rs

```diff
@@ -308,14 +310,106 @@ where
 #[cfg(test)]
 mod tests {
     use super::*;
+    use std::collections::hash_map::RandomState;
+
+    macro_rules! assert_empty {
+        ($x:expr) => {
+            assert!($x.is_empty());
+            assert_eq!($x.len(), 0);
```

Comment:
I would do the `assert_eq` on the length first, _then_ the check on `is_empty`. This is a bit more helpful when debugging, because if the assertion fails you're told what the length _actually_ was. Same for `assert_len` below.

Reply:
Good thinking! c907045225d49562e0e2c58bf8d4b4fc68350c02